### PR TITLE
:bug:[#1474] fix exception thrown with bad catalogus ID

### LINKED ISSUE #1474: Adding a ZaakType without specifying a Catalogus results raises an exception.
**General information**
Open Zaak version: dev

**Describe the bug**
Adding a ZaakType without specifying a Catalogus raise an RelatedObjectDoesNotExist exception instead of a validation error being shown.

**To Reproduce**
1. Install Open Zaak
2. Add a minimal ZaakType but leave "Catalogus" empty.
3. Press save.

**Expected behavior**
Instead of a RelatedObjectDoesNotExist (or probably an internal server error on production environments) I would expect clear error message to be shown indicating that I need to ad d Catalogus object.

Scherm­afbeelding 2023-09-12 om 15 03 50

### PR DESCRIPTION
**Pull request by Coperh**
Fixes #1474

**Changes**
Catch related object does not exist exception

### CODE REVIEW

File: src/openzaak/components/catalogi/admin/forms.py [hunk 1]
```diff
@@ -179,6 +180,12 @@ def clean(self):
         if "_publish" in self.data:
             self._clean_all_data_present_for_publish()
 
+        if not Catalogus.objects.filter(pk=self.data["catalogus"]).exists():
+            msg = _("Catalogus provided does not exist")
```
Comment (Coperh) [inline src/openzaak/components/catalogi/admin/forms.py:184]: What should this error be? What is the translation?

**Review by annashamray — CHANGES_REQUESTED** [review]

**Comment by annashamray** [comment]
@Coperh Could you please either resolve conversations or answer my comments, so I would know which comments are addressed?

File: src/openzaak/components/catalogi/models/mixins.py [hunk 2]
```diff
@@ -58,21 +58,23 @@ def clean(self):
...
+                begin_geldigheid=self.datum_begin_geldigheid,
+                einde_geldigheid=self.datum_einde_geldigheid,
+                instance=self,
+            ):
+                raise ValidationError(
+                    f"{self._meta.verbose_name} versies (dezelfde omschrijving) mogen geen "
+                    "overlappende geldigheid hebben."
+                )
+        except self.__class__.catalogus.RelatedObjectDoesNotExist:
+            raise ValidationError(_("Catalogus provided does not exist"))
```
[the remark this answers is not in the document]
Comment (Coperh) [reply src/openzaak/components/catalogi/models/mixins.py]: Suggestion implemented. 
Also removed Validation error.

File: src/openzaak/components/catalogi/tests/admin/test_zaaktype_admin.py [hunk 3]
```diff
@@ -625,6 +625,24 @@ def test_default_selectielijst_year_regression(self, m):
...
+        Test that a bad catalogus ID causes a validation error and not an exception.
+        Fixes #1474
+        """
+        url = reverse("admin:catalogi_zaaktype_add")
+        mock_selectielijst_oas_get(m)
+        mock_resource_list(m, "procestypen")
+        add_page = self.app.get(url)
+        form = add_page.form
+
+        form["catalogus"] = 16
```
[the remark this answers is not in the document]
Comment (Coperh) [reply src/openzaak/components/catalogi/tests/admin/test_zaaktype_admin.py:656]: Empty case is added.

**Comment by Coperh** [comment]
@annashamray yep, sorry.

That is done now.

**Review by annashamray — APPROVED** [review]

**Pull request merged**